llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -87,6 +106,23 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
                 self.remote_ud.init(state, logger)
                 state.callbacks.append(self.remote_ud)
 
+            if self.log_to_mlflow:
+                self.mlflow_loggers = [
+                    logger_destination
+                    for logger_destination in logger.destinations
+                    if isinstance(logger_destination, MLFlowLogger)
+                ]
+                if len(self.mlflow_loggers) == 0:
+                    raise ValueError(
+                        f'`log_to_mlflow` was set to `True` but no `MLFlowLogger` was found in the `logger.destinations` list. '
+                        +
+                        'Please add an `MLFlowLogger` or set `log_to_mlflow` to `False`.'
+                    )
+
+                import mlflow
+                mlflow.environment_variables.MLFLOW_HUGGINGFACE_MODEL_MAX_SHARD_SIZE.set(
+                    '5GB')
```

Comment:
I ran into issues too with the default MLflow HF shard size being too small (which slowed down file upload/download). Curious how 5GB worked for you - we might want to update the default in MLflow to something like 1GB at least, if we can confirm it won't regress/OOM any existing workloads

Reply:
yeah, 5GB seemed alright. HF's default is 10. any particular reason y'all are deviating from that?